src/carma-streets/parse_kafka_logs.py

```diff
@@ -124,7 +142,77 @@ def parse_timesync_to_csv(inputfile: Path, outputfile: Path):
             print('Finished writing all entries successfully')
         else:
             print(f'WARNING: Skipped {skipped_messages} due to errors. Please inspect logs')
-   
+
+def parse_map_to_csv(inputfile: Path, outputfile: Path):
+    """Function to parse MAP Kafka Topic log file and generate csv data of all time sync messages
+
+    Args:
+        inputfile (Path): Path to Kafka Topic log file
+        outputfile (Path): File name (excluding file extension) of desired csv file
+    """
+
+    map_msgs = parse_kafka_logs_as_type(inputfile, KafkaLogMessageType.MAP)
+    if  outputfile.exists():
+        print(f'Output file {outputfile} already exists. Overwritting file.')
+    #write data of interest to csv which will be used to produce plots
+    with open(outputfile, 'w', newline='') as write_obj:
```

Comment:
I normally see the `with open() as` idiom use `file` for the object name:

```python
with open(outputfile, 'w', newline='') as file:
```

Consider updating.